crates/stages/src/stages/sender_recovery.rs

```diff
@@ -162,16 +196,32 @@ impl<DB: Database> Stage<DB> for SenderRecoveryStage {
     }
 }
 
-// TODO(onbjerg): Should unwind
 #[derive(Error, Debug)]
+#[error(transparent)]
 enum SenderRecoveryStageError {
-    #[error("Sender recovery failed for transaction {tx}.")]
-    SenderRecovery { tx: TxNumber },
+    /// A transaction failed sender recovery
+    FailedRecovery(FailedSenderRecoveryError),
+
+    /// A different type of stage error occurred
+    StageError(#[from] StageError),
+}
+
+// TODO(onbjerg): Should unwind
+#[derive(Error, Debug)]
+#[error("Sender recovery failed for transaction {tx}.")]
+struct FailedSenderRecoveryError {
+    /// The transaction that failed sender recovery
+    tx: TxNumber,
 }
 
-impl From<SenderRecoveryStageError> for StageError {
-    fn from(error: SenderRecoveryStageError) -> Self {
-        StageError::Fatal(Box::new(error))
+impl FailedSenderRecoveryError {
+    /// Creates a [StageError] populated with a transaction sender recovery error with the given
+    /// block.
+    fn with_block(block: SealedHeader) -> StageError {
+        StageError::Validation {
+            block,
+            error: consensus::ConsensusError::TransactionSignerRecoveryError,
+        }
```

Comment:
This is weird, should `.into()` it into `FailedSenderRecoveryError`

Reply:
just removed `with_block` and returned the `StageError` when handling the failed recovery error